### PR TITLE
Install test suite for backend packages

### PR DESCRIPTION
**Pull request by sneumann**
This is working on #186 . Spectra now includes (and installs) a directory with tests that can be called from the backends. There could (should!) be a few different profiles, checking a (minimum) set of functionality provided by the backends. The initially provided one barely tests anything yet. Yours, Steffen

### CODE REVIEW

**Comment by jorainer** [comment]
This PR looks extremely interesting, thanks @sneumann and @hechth. I would however provide an already initialized `MsBackend` with the `be` variable since not all backends will have (only) a `file` parameter (e.g. expecting a database connection instead or even a data frame or similar).

Another very important test suite would be a general `test_valid_MsBackend` test suite that takes an already initialized `MsBackend` variable `be` and then runs all methods defined in the `MsBackend.R` file and checks that the returned result is as expected (i.e. expected data type, length of data matches expectation etc). Question is if you are planning to add that too or if we (I) should do that?

File: inst/test_msmslibraries/test_MsBackendMsmslibraries.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
+## assuming an object "be" exists with an initialises backend
```
Comment (jorainer) [inline inst/test_msmslibraries/test_MsBackendMsmslibraries.R:1]: Please add here a little more documentation/notes, such as
- what is expected (variable name(s), what they should be
- how can this test be called from another package
- what is being tested by the tests within this file

**Review by jorainer — CHANGES_REQUESTED** [review]
I think that's an excellent contribution, thanks @sneumann and @hechth. I made some comments and in general I think it would be better to structure the directories in a different way:

inst/test_backends/ being the base folder, and in this one having different folders e.g. your test_msmslibraries/ defining different test suites.

**Comment by sneumann** [comment]
I think this can now be closed in favour of https://github.com/rformassspectrometry/Spectra/pull/227 ?

**Comment by jorainer** [comment]
It's up to you @sneumann - you're more than welcome to add additional test suites. #227 adds *only* test suites for accessing/setting variables.

**Comment by jorainer** [comment]
Closing this PR now.

**Pull request closed without merging**